refactor(napi/oxlint): use `assertIs` function instead of type assertions

overlookmotel opened the pull request:
Pure refactor, and only affects type-checking, not runtime behavior.

Replace various `as` type assertions by using an `assertIs` function which performs the same role. #12620 enabled syntax minification in TSDown, so these `assertIs<...>(...)` calls are removed in the build. This gives us better typing without any runtime cost.

It seems to work - I've checked that `assertIs` is not present in the bundled code in `dist`.

camc314 commented:
### Merge activity

* **Jul 30, 12:31 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(napi/oxlint): use `assertIs` function instead of type assertions (#12621)

Pure refactor, and only affects type-checking, not runtime behavior.

Replace various `as` type assertions by using an `assertIs` function which performs the same role. #12620 enabled syntax minification in TSDown, so these `assertIs<...>(...)` calls are removed in the build. This gives us better typing without any runtime cost.

It seems to work - I've checked that `assertIs` is not present in the bundled code in `dist`.
